src/Controller/ProductController.php

```diff
@@ -103,120 +111,134 @@ public function getProductsAction(
+        EntityManagerInterface $manager,
+        ConstraintViolationListInterface $violations
+    ) {
         $this->handleViolations($violations);
 
         $manager->persist($newProduct);
         $manager->flush();
-        $view = $this->view(
-            $newProduct,
-            Response::HTTP_CREATED,
-            ['Location' => $this->generateUrl(
-                'product_show_id',
-                [
-                    'id' => $newProduct->getId(),
-                    UrlGeneratorInterface::ABSOLUTE_URL
-                ]
-            )]
-        );
 
-        return $this->handleView($view);
+        return $this->view($newProduct, Response::HTTP_CREATED);
     }
 
     /**
+     * Modify a phone (admin only)
+     *
      * @Post(
-     *     "/products/{id}",
+     *     "/api/products/{id}",
      *     name = "product_edit",
      *     requirements = {"id": "\d+"}
      * )
      * @ParamConverter("modifiedProduct", converter="fos_rest.request_body")
+     * @View()
+     * @IsGranted("ROLE_ADMIN")
+     * @SWG\Response(
+     *     response = 202,
+     *     description = "Update a product (admin only)"
+     * )
      */
-    public function editAction(Product $product, Product $modifiedProduct, EntityManagerInterface $manager)
-    {
+    public function editProductAction(
+        Product $product,
+        Product $modifiedProduct,
+        EntityManagerInterface $manager
+    ) {
         if ($modifiedProduct->getBrand() !== null) {
             $product->setBrand($modifiedProduct->getBrand());
         }
-
         if ($modifiedProduct->getModel() !== null) {
             $product->setModel($modifiedProduct->getModel());
         }
-
         if ($modifiedProduct->getPrice() !== null) {
             $product->setPrice($modifiedProduct->getPrice());
         }
-
         if ($modifiedProduct->getQuantity() !== null) {
             $product->setQuantity($modifiedProduct->getQuantity());
         }
 
         $manager->flush();
-        $view = $this->view($product, Response::HTTP_ACCEPTED);
 
-        return $this->handleView($view);
+        return $this->view($product, Response::HTTP_ACCEPTED);
     }
 
     /**
+     * Delete a phone (admin only)
+     *
      * @Delete(
-     *     "/products/{id}",
+     *     "/api/products/{id}",
      *     name = "product_delete"
      * )
+     * @View()
+     * @IsGranted("ROLE_ADMIN")
+     * @SWG\Response(
+     *     response = 200,
+     *     description = "Delete a product (admin only)"
+     * )
      */
-    public function deleteAction(Product $product, EntityManagerInterface $manager)
+    public function deleteProductAction(Product $product, EntityManagerInterface $manager)
     {
+        $id = $product->getId();
         $manager->remove($product);
         $manager->flush();
-        $view = $this->view($product, Response::HTTP_ACCEPTED);
 
-        return $this->handleView($view);
+        return  $this->view("Product {$id} deleted.", Response::HTTP_OK);
```

Comment:
ℹ️ In an api, we usually don't return any "notification" saying that the product was deleted. The HTTP code already says that

Reply:
ok